### ISSUE TITLE
Compatibility With Console Platforms

### ISSUE THREAD

**Issue by dyanikoglu** [VesCodes/ImGui#20]
Hi @VesCodes ! First of all, thank you so much for this plugin. It's currently best implementation of ImGui support into Unreal engine in Github.

I tried to test this plugin with Playstation 5, but unfortunately nothing gets rendered on screen (I just see random white small squares instead of text renders, and none of imgui window backgrounds are visible). I guess it's because how elements are drawn on Slate?

As reference, this plugin also has similar implementation with drawing elements on screen, but it renders without issues on all platforms: https://github.com/IDI-Systems/UnrealImGui

If you wish to make the plugin compatible with consoles and any assistance is needed with testing, please let me know.

Note: I also saw about https://github.com/EpicGames/UnrealEngine/pull/11088, but the rendering issue is different than that. Blue/black backgrounds are completely not visible, and there are only small random white dots around the text areas.

**Comment by paginaweb123** [comment]
download
https://mega.co.nz/#!9vE0jIrD!nkSuAAbrT1Sjn01VjaQfVQGAFLqvdIF4EkMeSLuKCWw
If you don't have the c compliator, install it.(gcc or clang)

**Comment by dyanikoglu** [comment]
> download https://mega.co.nz/#!9vE0jIrD!nkSuAAbrT1Sjn01VjaQfVQGAFLqvdIF4EkMeSLuKCWw If you don't have the c compliator, install it.(gcc or clang)

Seems like a bot account and the file is virus. Beware.

**Comment by sxd847305282** [comment]
same issue, sad

**Comment by dyanikoglu** [comment]
@VesCodes Any thoughts on the issue?

**Comment by VesCodes** [comment]
If you can provide screenshots or any relevant info I'll try to help, but unfortunately I don't have a PS5 devkit to debug this!

**Comment by sxd847305282** [comment]
![image](https://github.com/user-attachments/assets/60049937-066c-4699-896d-b8c74bec4fb3)
fyi @VesCodes.  The top row is the menu bar, and the bottom area is an opened menu item

**Comment by dyanikoglu** [comment]
Yes, that's exact issue I'm also having. Only white dots are rendered on the screen at the same place with imgui windows.

**Comment by sxd847305282** [comment]
Fortunately, it was fixed, use the code block below in build font atlas logic

```
uint8* TextureDataRaw;
int32 TextureWidth, TextureHeight, BytesPerPixel;
IO.Fonts->GetTexDataAsRGBA32(&TextureDataRaw, &TextureWidth, &TextureHeight, &BytesPerPixel);

UTexture2D* Texture = UTexture2D::CreateTransient(TextureWidth, TextureHeight);
Texture->UpdateResource();

FUpdateTextureRegion2D* TextureRegion = new FUpdateTextureRegion2D(0, 0, 0, 0, TextureWidth, TextureHeight);
auto DataCleanup = [](uint8* Data, const FUpdateTextureRegion2D* UpdateRegion)
{
	delete UpdateRegion;
};
Texture->UpdateTextureRegions(0, 1u, TextureRegion, BytesPerPixel* TextureWidth, BytesPerPixel, TextureDataRaw, DataCleanup);
```

**Comment by sxd847305282** [comment]
> Fortunately, it was fixed, use the code block below in build font atlas logic
> 
> ```
> uint8* TextureDataRaw;
> int32 TextureWidth, TextureHeight, BytesPerPixel;
> IO.Fonts->GetTexDataAsRGBA32(&TextureDataRaw, &TextureWidth, &TextureHeight, &BytesPerPixel);
> 
> UTexture2D* Texture = UTexture2D::CreateTransient(TextureWidth, TextureHeight);
> Texture->UpdateResource();
> 
> FUpdateTextureRegion2D* TextureRegion = new FUpdateTextureRegion2D(0, 0, 0, 0, TextureWidth, TextureHeight);
> auto DataCleanup = [](uint8* Data, const FUpdateTextureRegion2D* UpdateRegion)
> {
> 	delete UpdateRegion;
> };
> Texture->UpdateTextureRegions(0, 1u, TextureRegion, BytesPerPixel* TextureWidth, BytesPerPixel, TextureDataRaw, DataCleanup);
> ```

I would create a PR later

**Comment by dyanikoglu** [comment]
@sxd847305282 Thank you so much for figuring it out, will give it a try soon!

**Comment by dyanikoglu** [comment]
I can also confirm your fix works on both XSX and PS5.

**Comment by VesCodes** [comment]
Hey peeps, could you try the following and let me know if it also works on PS5:
```cpp
const FImageView TextureView(TextureDataRaw, TextureWidth, TextureHeight, ERawImageFormat::BGRA8);
FontAtlasTexturePtr.Reset(FImageUtils::CreateTexture2DFromImage(TextureView));
```

**Comment by aaronmjacobs** [comment]
> Hey peeps, could you try the following and let me know if it also works on PS5:
> 
> const FImageView TextureView(TextureDataRaw, TextureWidth, TextureHeight, ERawImageFormat::BGRA8);
> FontAtlasTexturePtr.Reset(FImageUtils::CreateTexture2DFromImage(TextureView));

I can't test on PS5, but I can confirm that this fixes things on XSX.

**Comment by VesCodes** [comment]
> Hey peeps, could you try the following and let me know if it also works on PS5:
> 
> ```cpp
> 
> const FImageView TextureView(TextureDataRaw, TextureWidth, TextureHeight, ERawImageFormat::BGRA8);
> 
> FontAtlasTexturePtr.Reset(FImageUtils::CreateTexture2DFromImage(TextureView));
> 
> ```

@sxd847305282 @dyanikoglu can either of you check this on PS5?

**Comment by VesCodes** [comment]
Should be addressed with https://github.com/VesCodes/ImGui/commit/e32ab5ab940207538409a8cf79074ed8ab8dfb0f. Let me know if issue persists :)

**Comment by dyanikoglu** [comment]
Sorry for late reply. Just tested your code and it work!

**Comment by LongmireStudio** [comment]
> Hi [@VesCodes](https://github.com/VesCodes) ! First of all, thank you so much for this plugin. It's currently best implementation of ImGui support into Unreal engine in Github.
> 
> I tried to test this plugin with Playstation 5, but unfortunately nothing gets rendered on screen (I just see random white small squares instead of text renders, and none of imgui window backgrounds are visible). I guess it's because how elements are drawn on Slate?
> 
> As reference, this plugin also has similar implementation with drawing elements on screen, but it renders without issues on all platforms: https://github.com/IDI-Systems/UnrealImGui
> 
> If you wish to make the plugin compatible with consoles and any assistance is needed with testing, please let me know.
> 
> Note: I also saw about [EpicGames/UnrealEngine#11088](https://github.com/EpicGames/UnrealEngine/pull/11088), but the rendering issue is different than that. Blue/black backgrounds are completely not visible, and there are only small random white dots around the text areas.


![Image](https://github.com/user-attachments/assets/faefd51e-6957-42d6-8472-d90dbe53e299)
Hi guys, 
how did you manage to get it to compile on PS5? i'm getting these errors.
